packages/components/src/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -0,0 +1,394 @@
+        model = model.merge({domain:newDomain}) as SampleTypeModel;
+        this.setState(() => ({
+            model,
+            invalidDomainField
+        }));
+    };
+
+    renderDomainPanel = () => {
+        const {noun, domainDesignerStickyHeight} = this.props;
+        const {model} = this.state;
+        const {domain} = model;
+
+        return (
+            <>
+                {domain &&
+                <DomainForm
+                        domain={domain}
+                        onChange={this.domainChangeHandler}
+                        showHeader={true}
+                        initCollapsed={true}
+                        collapsible={true}
+                        helpNoun={noun.toLowerCase()}
+                        useTheme={false}
+                        appPropertiesOnly={true}
+                        containerTop={domainDesignerStickyHeight}
+                />
+                }
+            </>
+        );
+    };
+
+    isFormValid = (): boolean => {
+        return SampleTypeModel.isValid(this.state.model);
+    };
+
+    onFinish = () => {
+        const { beforeFinish } = this.props;
+        const { model } = this.state;
+
+        this.setSubmitting(true);
+
+        if (beforeFinish) {
+            beforeFinish(model);
+        }
+
+        const {name, domain, description, nameExpression } = model;
+
+        let domainDesign = domain.merge({
+            name: name, //This will be the Sample Type Name
+            description: description,
+        }) as DomainDesign;
+
+        const details = {
+            name,
+            nameExpression,
+            importAliases: SampleTypeDesigner.getImportAliasesAsMap(model).toJS(),
+        };
+
+        if (model.isNew())
+        {
+            //Initialize a sampleId column, this is not displayed as part of the designer.
+            const nameCol = {
+                name: 'Name'
+            };
+
+            domainDesign = addDomainField(domainDesign, nameCol);
+        }
+
+        saveDomain(domainDesign, KINDS.SAMPLE_TYPE, details, name)
+            .then((savedDomain) => {
+                this.onFinishSuccess(savedDomain);
+            })
+            .catch((errorDomain) => {
+                this.onFinishFailure(resolveErrorMessage(errorDomain));
+            });
+
+    };
+
+    onFinishSuccess(response: any) {
+        this.setSubmitting(false);
+        this.props.onComplete(response);
+    }
+
+    onFinishFailure(error: React.ReactNode) {
+        this.setState(() => ({
+            error,
+            submitting: false
+        }));
+    }
+
+    setSubmitting(submitting: boolean) {
+        this.setState(() => ({
+            error: undefined,
+            submitting
+        }));
+    }
+
+    render() {
+        const { onCancel } = this.props;
+        const { submitting, error, model } = this.state;
+        if (!model)
+            return null;
+
+        return (
+            <>
+                {error && <Alert>{error}</Alert>}
```

Comment:
I believe we have usually been putting the error alert between the panels and the buttons so they show closer to where the user clicks the save button.

Reply:
moved.